nikola/plugin_categories.py

```diff
@@ -201,10 +200,14 @@ class TemplateSystem(BasePlugin):
 
     name = "dummy_templates"
 
-    def set_directories(self, directories: 'typing.List[str]', cache_folder: str):
+    def set_directories(self, directories: List[str], cache_folder: str) -> None:
         """Set the list of folders where templates are located and cache."""
         raise NotImplementedError()
 
+    def user_configuration(self, user_config: Mapping[str, str]) -> None:
```

Comment:
Consider adding a verb to the method name, perhaps `apply_user_configuration`? (`Apply` would also work nicely in the docstring.)